src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -451,12 +451,18 @@ public static bool IsErrorRedirectedCore()
             return IsHandleRedirected(Interop.Sys.FileDescriptors.STDERR_FILENO);
         }
 
+        private static Encoding s_consoleEncoding;
+
         /// <summary>Creates an encoding from the current environment.</summary>
         /// <returns>The encoding.</returns>
         private static Encoding GetConsoleEncoding()
         {
-            Encoding enc = EncodingHelper.GetEncodingFromCharset();
-            return enc ?? new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+            if (s_consoleEncoding == null)
+            {
+                Encoding enc = EncodingHelper.GetEncodingFromCharset();
```

Comment:
This should not be cached.  It can be changed via calls to Environment.SetEnvironmentVariable.